html2/q2.js

```diff
@@ -0,0 +1,156 @@
+const movies=[
+    
+  {
+    "MovieName": "The Great Adventure",
+    "ActorName": "John Smith",
+    "ReleaseDate": "2023-01-15"
+  },
+  {
+    "MovieName": "Mystery in the Woods",
+    "ActorName": "Emily Johnson",
+    "ReleaseDate": "2022-09-28"
+  },
+  {
+    "MovieName": "Love and Destiny",
+    "ActorName": "Michael Brown",
+    "ReleaseDate": "2023-05-02"
+  },
+  {
+    "MovieName": "City of Shadows",
+    "ActorName": "Sophia Williams",
+    "ReleaseDate": "2023-03-12"
+  },
+  {
+    "MovieName": "The Last Stand",
+    "ActorName": "William Davis",
+    "ReleaseDate": "2022-11-07"
+  },
+  {
+    "MovieName": "Echoes of Time",
+    "ActorName": "Olivia Wilson",
+    "ReleaseDate": "2022-12-19"
+  }
+
+];
+
+//1. List the movie name along with the actor name of those movies released in the year 2022
+console.log(movies
+    .filter(x=> new Date(x.ReleaseDate).getFullYear() === 2022)
+    .map(x=>`Movie : ${x.MovieName} , Actor : ${x.ActorName}`));
+
+//2. List the movie names released in the year 2023 where the actor is William Davis.
+console.log(movies
+    .filter(x=> x.ActorName=='William Davis' && new Date(x.ReleaseDate).getFullYear() === 2023)
+    .map(x=>`Movie : ${x.MovieName} , Actor : ${x.ActorName}`));
+
+//3. Retrieve the Actor name and release date of the movie “The Last Stand”
+console.log(movies
+    .filter(x=>x.MovieName== 'The Last Stand')
+    .map(x=>`Actor : ${x.ActorName} , Release : ${x.ReleaseDate}`));
+
+//4. Check whether there is any movie in the list with actor name “John Doe”
+console.log(movies
+    .some(x=>x.ActorName=='John Doe'));
+
+//5. Display the count of movies where the actor name is "Sophia Williams"
+console.log(movies
+    .filter(x=>x.ActorName=='Sophia Williams').length);
+
+// 6. Insert an element
+// {
+//     "MovieName": "The Final Stage",
+//     "ActorName": "John Doe",
+//     "ReleaseDate": "2022-08-11"
+// }
+// as last element
+
+const entry={
+
+    "MovieName": "The Final Stage",
+    "ActorName": "John Doe",
+    "ReleaseDate": "2022-08-11"
+}
+
+movies.push(entry);
+movies.forEach(x => {
+    console.log(x)
+});
+
+//7. Check whether there exists any duplicate movie names present in the array
+const titles=movies.filter(x=>x.MovieName)
```

Comment:
retest whether this work as you would expect

Reply:
There was a mistake. Used map to produce the output correctly

const titles=movies.map(x=>x.MovieName)
console.log(titles);
const dup=new Set(titles);
console.log(dup.size)
console.log(movies.length)
console.log(dup.size<movies.length?'Has duplicates':"no duplicates")